configure

```diff
@@ -74,6 +74,11 @@ set_melang_default_paths() {
 }
 
 set_c_compilers() {
+    if [ -n "$CC" ]; then
+        cc="$CC"
+        tcc=${TCC:-$CC}
+        return
```

Comment:
There is a parameter named `--cc` for this `configure` to allow developer to set c compiler.

And it shouldn't be returned here because it still need to check the environment for Mac M1 and M2 later.

Reply:
Fixed the early `return` problem.

I see the `--cc` parameter, but I still think it will be better if we support GNU Environment Variables, because
> Since gnu env vars are widely understood, using them would lower the entry barrier plus would make the life of package maintainers simpler (from [a bazel issue](https://github.com/bazelbuild/bazel/issues/5186))